### PR TITLE
Require a matching regular operator for a checked operator

### PR DESCRIPTION
**Pull request by AlekseyTs**
https://github.com/dotnet/csharplang/blob/main/proposals/checked-user-defined-operators.md

### CODE REVIEW

File: src/Compilers/CSharp/Test/Symbol/Symbols/CheckedUserDefinedOperatorsTests.cs [hunk 1]
```diff
@@ -2668,6 +2741,29 @@ static class C
...
+
+    public static explicit operator C(int x) => default;
+    public static explicit operator int(C? x) => default;
+    public static explicit operator long(C x) => default;
+}
+";
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugDll, parseOptions: TestOptions.RegularPreview);
+
+            compilation1.VerifyDiagnostics(
+                // (4,45): error CS0216: The operator 'C.explicit operator checked int(C)' requires a matching operator 'explicit' to also be defined
```
Comment (chsienki) [inline src/Compilers/CSharp/Test/Symbol/Symbols/CheckedUserDefinedOperatorsTests.cs:2761]: This error message seems weird? I would expect to be told which operator I need to actually define.

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Symbol/Symbols/CheckedUserDefinedOperatorsTests.cs:2761]: > This error message seems weird? I would expect to be told which operator I need to actually define.

Could you give specific example given your understanding of the rules?

**Comment by AlekseyTs** [comment]
@333fred, @chsienki, @dotnet/roslyn-compiler Please review.

**Review by chsienki — APPROVED** [review]

File: src/Compilers/CSharp/Test/Symbol/Symbols/CheckedUserDefinedOperatorsTests.cs (hunk 1 above)
Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Symbol/Symbols/CheckedUserDefinedOperatorsTests.cs:2761]: I will add a PROTOTYPE comment to follow up on the wording.

**Comment by AlekseyTs** [comment]
@333fred, @dotnet/roslyn-compiler For the second review

**Comment by AlekseyTs** [comment]
@333fred, https://github.com/orgs/dotnet/teams/roslyn-compiler For the second review

File: src/Compilers/CSharp/Test/Symbol/Symbols/CheckedUserDefinedOperatorsTests.cs (hunk 1 above)
Comment (333fred) [reply src/Compilers/CSharp/Test/Symbol/Symbols/CheckedUserDefinedOperatorsTests.cs:2761]: I think the thing throwing me is that the issue is with the lack of an _unmodified_ operator. I read this error message and go "I already have an explicit conversion operator, what else do you want from me?". As a member of the language design team I know what that means, but I don't expect it to be immediately obvious to most users.

**Review by 333fred — APPROVED** [review]
LGTM (commit 2)

**Pull request merged**